### PR TITLE
Files with the same name #114

### PR DESCRIPTION
**Pull request by sholkay**
@Swati4star Please review.

- I have just implemented a validation point where it will check if the file with the same is already there or not. If the file exists then another dialogue will pop up otherwise new file will be created with that name.

### CODE REVIEW

**Comment by Swati4star** [comment]
For any issue, that has some UI change, please do add the screenshots also here. It becomes easier to review.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -64,6 +64,9 @@
     <string name="snackbar_no_pdf_app">No app to read PDF File</string>
     <string name="open_file">Open File</string>
     <string name="sort">Sort</string>
+    <string name="file_already_exists">Files with the same name already exists. Do you want to overwrite the existing file?\nPress No, to change the name.</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: Change `Files` to `File`.

Remove `\nPress No, to change the name.`. It is kind of creating a confusion.

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 2]
```diff
@@ -196,7 +195,34 @@ public void onInput(@NonNull MaterialDialog dialog, CharSequence input) {
...
+                                            @Override
+                                            public void onClick(MaterialDialog dialog, DialogAction which) {
+                                                new CreatingPdf().execute();
+                                            }
+                                        })
+                                        .negativeText(R.string.no)
+                                        .onNegative(new MaterialDialog.SingleButtonCallback() {
+                                            @Override
+                                            public void onClick(MaterialDialog dialog, DialogAction which) {
+                                                createPdf();
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: You should finish the current dialog also.

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 3]
```diff
@@ -196,7 +195,34 @@ public void onInput(@NonNull MaterialDialog dialog, CharSequence input) {
...
 
-                            new CreatingPdf().execute();
+                            mPath = Environment.getExternalStorageDirectory().getAbsolutePath() +
+                                    HomeFragment.this.getString(R.string.pdf_dir);
+
+                            mPath = mPath + mFilename + HomeFragment.this.getString(R.string.pdf_ext);
+
+                            File tempFile = new File(mPath);
+                            if (tempFile.exists()) {
+                                new MaterialDialog.Builder(mActivity)
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: Move this code (creating the dialog) to a new function. This function is becoming too complex.

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 4]
```diff
@@ -196,7 +195,34 @@ public void onInput(@NonNull MaterialDialog dialog, CharSequence input) {
...
                             mFilename = input.toString();
 
-                            new CreatingPdf().execute();
+                            mPath = Environment.getExternalStorageDirectory().getAbsolutePath() +
+                                    HomeFragment.this.getString(R.string.pdf_dir);
+
+                            mPath = mPath + mFilename + HomeFragment.this.getString(R.string.pdf_ext);
+
+                            File tempFile = new File(mPath);
+                            if (tempFile.exists()) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: Add new function in `https://github.com/Swati4star/Images-to-PDF/blob/master/app/src/main/java/swati4star/createpdf/util/FileUtils.java` for checking if a file exists & call that here.

**Comment by Swati4star** [comment]
You have taken the other commits (apart from yours) also. See the changed files.

**Comment by sholkay** [comment]
I messed up again. Wait, i'll fix it.

**Comment by Swati4star** [comment]
@sholkay Are you working on this ?

**Comment by Swati4star** [comment]
Closing because of no response!

**Pull request closed without merging**